app/assets/javascripts/views/user_form.js

```diff
@@ -0,0 +1,121 @@
+App.Views.UserForm = App.Views.BootstrapModal.extend({
+  template: JST['templates/forms/user/form'],
+
+  events: {
+    'click button[name=close]': 'onClose',
+    'click button[name=save]':  'onSave',
+  },
+
+  prepareData: function() {
+    this.content = this.template();
+    this.buttons = '';
+    this.buttons += JST['templates/forms/user/buttons/close']();
+    this.buttons += JST['templates/forms/user/buttons/save']();
+
+    this.model = this.options.user;
+    this.model.isChanged = false;
+
+    this.collection = this.options.collection;
+  },
+
+  addDOMBindings: function() {
+    App.Helpers.assignDatePicker(this.$('[name=birth_date]'), {
+      startDate: '1900-01-01',
+      endDate: '2050-01-01',
+    });
+
+    App.Helpers.assignDatePicker(this.$('[name=employment_date]'));
+  },
+
+  addListeners: function() {
+    this.listenTo(this.model, 'change', this.onChange);
+    this.listenTo(this.model, 'error', this.onError);
+    this.listenTo(this.model, 'sync', this.onSuccess);
+    this.listenTo(this.model, 'invalid', this.onInvalid);
+  },
+
+  renderAssistant: function() {
+    this.setFormValues();
+  },
+
+  getFormValues: function() {
+    _.each(this.$('input[type=text]'), function(input) {
+      this.model.set(input.name, input.value.trim());
+    }, this);
+  },
+
+  setFormValues: function() {
+    _.each(this.$('input[type=text]'), function(input) {
+      input.value = this.model.get(input.name);
+    }, this);
+  },
+
+  onClear: function() {
+    this.clearModel();
+    this.clearForm();
+  },
+
+  onSave: function() {
+    var isValid = false;
+
+    this.getFormValues();
+    isValid = this.model.isValid(true);
+
+    if (isValid && this.model.isChanged) {
+      this.model.save();
+    } else if (isValid && !this.model.isChanged) {
+      this.hide();
+    } else {
+      console.log('onSave: unpredictable state');
+    }
+  },
+
+  onClose: function(event) {
+    this.hide();
+  },
+
+  onChange: function(model, options) {
+    this.model.isChanged = true;
+  },
+
+  onError: function(model, response, options) {
+    var message = _.chain(response.responseJSON.errors)
```

Comment:
It looks like 'response.responseJSON.errors' is already is an array so this construction is useless.